geomesa-features/geomesa-feature-kryo/src/main/scala/org/locationtech/geomesa/features/kryo/KryoBufferSimpleFeature.scala

```diff
@@ -91,27 +118,14 @@ class KryoBufferSimpleFeature(sft: SimpleFeatureType,
     * @param length number of valid bytes to read from the byte array
     */
   def setBuffer(bytes: Array[Byte], offset: Int, length: Int): Unit = {
-    this.offset = offset
-    this.length = length
     input.setBuffer(bytes, offset, length)
-    // reset our offsets
-    input.setPosition(offset + 1) // skip version
-    startOfOffsets = offset + input.readInt()
-    input.setPosition(startOfOffsets) // set to offsets start
-    var i = 0
-    while (i < offsets.length && input.position < input.limit) {
-      offsets(i) = offset + input.readInt(true)
-      i += 1
-    }
-    if (i < offsets.length) {
-      // attributes have been added to the sft since this feature was serialized
-      missingAttributes = true
-      do { offsets(i) = -1; i += 1 } while (i < offsets.length)
-    } else {
-      missingAttributes = false
+    delegate = input.readByte() match {
+      case KryoFeatureSerializer.Version  => delegateV3
```

Comment:
Should Version 3 be named Version3?

Reply:
we had kept `Version` pointing to the most recent version (it used to be 2), but yeah I can see it either way. what do you think?